analysis.py

```diff
@@ -259,20 +301,27 @@ def parse_args() -> Namespace:
 def main() -> None:
     args = parse_args()
 
-    stripped_content, nodes_glob = read_raw_content(args.input_file[0])
+    stripped_content, run_number = read_raw_content(args.input_file[0])
 
     log_path = os.path.dirname(args.input_file[0])
 
-    node_logs = open_node_logs(os.path.join(log_path, nodes_glob))
+    node_logs = open_node_logs(os.path.join(log_path, f"node_{run_number}_*/*.log*"))
     filled_rows = fill_rows(stripped_content, node_logs)
     summary = generate_statistics(filled_rows)
+    output_directory = os.path.join(log_path, f"analysis_{run_number}")
+
+    if not os.path.exists(output_directory):
+        os.makedirs(output_directory)
```

Comment:
You can skip the if with
```suggestion
    os.makedirs(output_directory, exist_ok=True)
```